src/coreclr/jit/importer.cpp

```diff
@@ -4441,15 +4441,6 @@ bool Compiler::impIsImplicitTailCallCandidate(
         return false;
     }
 
-    // We cannot tailcall ValueTask returning methods as we need to preserve
-    // the Continuation instance for ValueTaskSource handling (the BCL needs
-    // to look at continuation.Next). We cannot easily differentiate between
-    // ValueTask and Task here, so we just disable it more generally.
-    if ((prefixFlags & PREFIX_IS_TASK_AWAIT) != 0)
-    {
-        return false;
-    }
-
```

Comment:
This check is moved to the same place as the check for `CORINFO_FLG_SYNCH`, since it is similar in spirit (we need to introduce try-finally, so we cannot tailcall).